### PR TITLE
Various VV work and fixes

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Refactored various procs used by VV: Header generation, dropdown menu generation, moved various things around, some defines.
Fixed VV to be able to edit associative lists.
Fixed VV to be able to make a list ex nihilo on a var.
Removed various dropdown menus that didn't exist anymore and were doing nothing.
Fixed add verb option to not wipe the already existing verbs of the mob.
Fixed a runtime error on some lists view building that were not handled correctly.
Fixed an issue with build mode not having the correct permissions for devs.
Fixed an issue with editing a list referenced by a key in another list via the edit referenced object option.
Added DF_VAR_EDITED flag.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

**Review by DreamySkrell — APPROVED** [review]

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -747,7 +747,7 @@ var/list/admin_verbs_cciaa = list(
 	set name = "Toggle Build Mode Self"
 	set category = "Special Verbs"
 
-	if(!check_rights(R_ADMIN))
+	if(!check_rights(R_ADMIN|R_DEV))
```
Comment (Arrow768) [inline code/modules/admin/admin_verbs.dm:750]:
```suggestion
	if(!check_rights(R_ADMIN|R_SPAWN))
```
Spawning is restricted to R_SPAWN

**Review by Arrow768 — DISMISSED** [review]

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (FluffyGhoster) [reply code/modules/admin/admin_verbs.dm:750]: This isn't spawning, this is the build mode, what was R_DEV before the update too?

Comment (Arrow768) [reply code/modules/admin/admin_verbs.dm:750]: AFAIK R_DEV has been removed quite a while from buildmode and moved to R_SPAWN given that the primary purpose of buildmode is to efficiently spawn stuff.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**